chore: Cleaning up the Bitcoin integration and ckBTC documentation (#5576)
dfinity/portal: docs/references/bitcoin-how-it-works.mdx

Canisters themselves can have ECDSA and Schnorr keys using novel chain-key ECDSA and Schnorr signature protocols and therefore can receive and hold bitcoin and other assets. Canisters can also create Bitcoin transactions and submit them via the Bitcoin API to the Bitcoin network. They use the chain-key ECDSA and Schnorr functionality to request threshold signatures to spend UTXOs in a transaction that is then submitted to the Bitcoin network. Chain-key signatures are much more than just a threshold ECDSA and threshold Schnorr implementation, as they, for example, also comprise protocols for secure distributed key generation and key rotation, which are crucial from a systems perspective to make threshold signing secure and practically viable. Details regarding ICP's chain-key signature protocol can be found on the [corresponding documentation page](/docs/references/t-sigs-how-it-works/).

The chain-key signature protocols expose an API on the management canister, whereas the Bitcoin integration endpoints are made available through the [Bitcoin canister](https://github.com/dfinity/bitcoin-canister).

> The APIs are low-level APIs designed around the concepts of Bitcoin UTXOs and transactions, are nontrivial to use, and require an in-depth understanding of how Bitcoin works. The chain-key ECDSA and Schnorr signature APIs are also generically useful for any use case, e.g., integration with other blockchains such as Ethereum.

Reviewer: ```suggestion
The low-level APIs are designed around the concepts of Bitcoin UTXOs and transactions, are nontrivial to use, and require an in-depth understanding of how Bitcoin works. The chain-key ECDSA and Schnorr signature APIs are also generically useful for any use case, e.g., integrations with other blockchains such as Ethereum.
```

> This page gives a high-level overview of the above-mentioned technology behind the direct Bitcoin integration. For details, the reader is referred to the [Bitcoin page on the Internet Computer wiki](https://wiki.internetcomputer.org/wiki/Bitcoin_integration) as well as the [threshold signatures documentation page](/docs/references/t-sigs-how-it-works/).

Reviewer: ```suggestion
This page gives a high-level overview of the above-mentioned technology behind the direct Bitcoin integration. For details, the reader is referred to the [Bitcoin page on the Internet Computer Learn Hub](https://learn.internetcomputer.org/hc/en-us/articles/34211154520084-Bitcoin-Integration) as well as the [threshold signatures documentation page](/docs/references/t-sigs-how-it-works/).
```

> This page gives a high-level overview of the above-mentioned technology behind the direct Bitcoin integration. For details, the reader is referred to the [Bitcoin page on the Internet Computer wiki](https://wiki.internetcomputer.org/wiki/Bitcoin_integration) as well as the [threshold signatures documentation page](/docs/references/t-sigs-how-it-works/).

Reviewer: Should the Wiki link be replaced with a page from the Learn Hub?

Author: Well, the Learn Hub is supposed to eventually replace the wiki, right?
If the Learn Hub page is already there, we can point to it now, of course.

Reviewer: Yes, there is a Learn Hub page, but I'm not sure if it conveys the same content as the Wiki page: https://learn.internetcomputer.org/hc/en-us/articles/34211154520084-Bitcoin-Integration

I believe the Wiki is meant to be deprecated and replaced by the Learn Hub, but I'm personally not sure of the timeline for that.

Author: The Learn Hub page actually has a link to the wiki page, so we can use the Learn Hub page here already now. If you agree, then feel free to propose the change - and start the merge process.   :slightly_smiling_face:

-   The **IC SDK** for local development of canisters.
-   ICP support as the pre-production environment for final testing on **Bitcoin testnet** (v4).
-   ICP support as the production environment for the release using the **Bitcoin mainnet**.

### Local SDK

In the typical canister development workflow, canisters on ICP are compiled and run in the local environment using the [IC SDK](/docs/building-apps/getting-started/install) during their development. Thus, the IC SDK is the first stage, or environment, of the development workflow. The IC SDK has been enabled to support both the Bitcoin integration and threshold ECDSA APIs.

In contrast to ICP deployments of the feature, which integrate with Bitcoin testnet and Bitcoin mainnet, respectively, the SDK integrates with a locally running `bitcoind` node in regression testing (regtest) mode. Using `bitcoind` in regtest mode is the preferred way for Bitcoin development. The Internet Computer has integrated the IC SDK with `bitcoind` in regtest mode to bring the best Bitcoin development experience to ICP. Both development and automated testing of smart contracts are first done in the local environment with this setup.

> The Bitcoin adapter of the single replica running the local SDK environment connects to the local `bitcoind` node instead of multiple nodes on the Bitcoin testnet or mainnet. Please look at the output of `dfx start --help` to see the relevant flags on dfx.

Reviewer: ```suggestion
The Bitcoin adapter of the single replica running the local SDK environment connects to the local `bitcoind` node instead of multiple nodes on the Bitcoin testnet or mainnet. Please look at the output of `dfx start --help` to see the relevant flags for `dfx`.
```

### Bitcoin testnet on ICP

Once a smart contract is ready for acceptance testing, it is deployed on the ICP mainnet (recall, there is no public ICP testnet), using the Bitcoin API set to connect to the Bitcoin testnet, i.e., no real value is at stake.

> This is the same setting as used in local development.

Reviewer: ```suggestion
This is the same setting used in local development.
```
